### ISSUE TITLE
Add functionality to add "Debug handlers" to the channel pipelines

### ISSUE THREAD

**Issue by Lukasa** [swift-server/async-http-client#596]
A useful trick we learned in grpc-swift is that it can be handy to allow users to insert a "debug handler" (or several) into each channel pipeline. This should usually be given the option to insert itself anywhere in the pipeline (meaning we should probably name our handlers), and can enable all kinds of useful observability and testing. We may need to be careful for HTTP/2, and we should consider @glbrntt's grpc experience here.

**Comment by weissi** [comment]
Yes please, I use that functionality pretty much on every gRPC server/client I've ever created :).
